Nil pointer dereference at concurent logging

When i use two logger instance concurrently then i get "invalid memory address or nil pointer dereference"
Here a sample code to reproduce the error
```go
package main
import (
	graylog "github.com/gemnasium/logrus-graylog-hook/v3"
	"github.com/sirupsen/logrus"
	"sync"
)
func main() {
	var wg sync.WaitGroup

	var gelfHook = graylog.NewGraylogHook("xx.xx.xx.xx:xxx", map[string]interface{}{})

	log1 := logrus.New()
	log2 := logrus.New()

	log1.AddHook(gelfHook)
	log2.AddHook(gelfHook)

	for i := 0; i < 100; i++ {
		wg.Add(1)
		go func() {
			defer wg.Done()
			log1.Info("test log info")
			log2.Info("test log info")
		}()
	}
	wg.Wait()
}

```
and the error message
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4d19da]

goroutine 23 [running]:
compress/flate.(*Writer).Close(...)
        /usr/lib/go-1.13/src/compress/flate/deflate.go:729
compress/gzip.(*Writer).Close(0xc0000be210, 0xc0004801e0, 0x97)
        /usr/lib/go-1.13/src/compress/gzip/gzip.go:248 +0x6a
github.com/gemnasium/logrus-graylog-hook/v3.(*Writer).WriteMessage(0xc0000d0070, 0xc00041a100, 0xc000141ac0, 0x0)
        /xxxx/go/pkg/mod/github.com/gemnasium/logrus-graylog-hook/v3@v3.0.2/gelf_writer.go:246 +0x2e4
github.com/gemnasium/logrus-graylog-hook/v3.(*GraylogHook).sendEntry(0xc0000ba120, 0xc000119340, 0x0, 0x0, 0x0)
        /xxxx/go/pkg/mod/github.com/gemnasium/logrus-graylog-hook/v3@v3.0.2/graylog_hook.go:247 +0xa2b
github.com/gemnasium/logrus-graylog-hook/v3.(*GraylogHook).Fire(0xc0000ba120, 0xc0001192d0, 0x0, 0x0)
        /xxxx/go/pkg/mod/github.com/gemnasium/logrus-graylog-hook/v3@v3.0.2/graylog_hook.go:120 +0x294
github.com/sirupsen/logrus.LevelHooks.Fire(0xc00009c750, 0x4, 0xc0001192d0, 0xc0001192d0, 0x40d3a4)
        /xxxx/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/hooks.go:28 +0x91
github.com/sirupsen/logrus.(*Entry).fireHooks(0xc0001192d0)
        /xxxx/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:246 +0xa3
github.com/sirupsen/logrus.Entry.log(0xc0000d0150, 0xc0001104e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /xxxx/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:224 +0xf4
github.com/sirupsen/logrus.(*Entry).Log(0xc000119260, 0xc000000004, 0xc000141f98, 0x1, 0x1)
        /xxxx/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:268 +0xeb
github.com/sirupsen/logrus.(*Logger).Log(0xc0000d0150, 0x4, 0xc000141f98, 0x1, 0x1)
        /xxxx/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/logger.go:192 +0x7d
github.com/sirupsen/logrus.(*Logger).Info(...)
        /xxxx/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/logger.go:206
main.main.func1(0xc0000b4250, 0xc0000d00e0, 0xc0000d0150)
        /xxxx/main.go:25 +0x110
created by main.main
        /xxxx/main.go:22 +0x26e
```

I think that something will be override because if add this lines 
```go
w.mu.Lock()
defer w.mu.Unlock()
```
to gelf_writer.go line 198 then everything work as expected

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,10 +1,7 @@
 language: go
 go:
-  - "1.8"
-  - "1.9"
-  - "1.10"
-  - "1.11"
-  - "tip"
+  - "stable"
+  - "master"
 
 env:
   - GO111MODULE=on

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Logrus Graylog hook
 
+## 3.0.3 - 2019-12-28
+
+* Fix concurrent logging when hook is reused (#49)
+
 ## 3.0.2 - 2019-01-10
 
 * TRACE level logs as syslog `LOG_DEBUG` level (7)

--- a/gelf_writer.go
+++ b/gelf_writer.go
@@ -195,6 +195,9 @@ type writerCloserResetter interface {
 // filled out appropriately.  In general, clients will want to use
 // Write, rather than WriteMessage.
 func (w *Writer) WriteMessage(m *Message) (err error) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
 	mBytes, err := json.Marshal(m)
 	if err != nil {
 		return

--- a/go.mod
+++ b/go.mod
@@ -4,3 +4,5 @@ require (
 	github.com/pkg/errors v0.8.1
 	github.com/sirupsen/logrus v1.3.0
 )
+
+go 1.13

--- a/graylog_hook_test.go
+++ b/graylog_hook_test.go
@@ -9,8 +9,8 @@ import (
 	"regexp"
 	"runtime"
 	"strings"
+	"sync"
 	"testing"
-	"time"
 
 	pkgerrors "github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
@@ -160,52 +160,49 @@ func TestParallelLogging(t *testing.T) {
 	log.Hooks.Add(hook)
 	log.Hooks.Add(asyncHook)
 
+	log2 := logrus.New()
+	log2.Out = ioutil.Discard
+	log2.Hooks.Add(hook)
+	log2.Hooks.Add(asyncHook)
+
 	quit := make(chan struct{})
 	defer close(quit)
 
-	panicked := false
-
 	recordPanic := func() {
 		if r := recover(); r != nil {
-			panicked = true
+			t.Fatalf("Logging in parallel caused a panic")
 		}
 	}
 
-	go func() {
-		// Start draining messages from GELF
-		go func() {
-			defer recordPanic()
-			for {
-				select {
-				case <-quit:
-					return
-				default:
-					r.ReadMessage()
-				}
-			}
-		}()
+	var wg sync.WaitGroup
 
-		// Log into our hook in parallel
-		for i := 0; i < 10; i++ {
-			go func() {
-				defer recordPanic()
-				for {
-					select {
-					case <-quit:
-						return
-					default:
-						log.Info("Logging")
-					}
-				}
-			}()
+	// Start draining messages from GELF
+	go func() {
+		defer recordPanic()
+		for {
+			select {
+			case <-quit:
+				return
+			default:
+				r.ReadMessage()
+			}
 		}
 	}()
 
-	// Let them all do their thing for a while
-	time.Sleep(100 * time.Millisecond)
-	if panicked {
-		t.Fatalf("Logging in parallel caused a panic")
+	// Log into our hook in parallel
+	for i := 0; i < 100; i++ {
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+			defer recordPanic()
+
+			log.Info("Logging")
+			log2.Info("Logging from another logger")
+		}()
 	}
+
+	wg.Wait()
 }
 
 func TestSetWriter(t *testing.T) {
@@ -271,7 +268,7 @@ func TestStackTracer(t *testing.T) {
 			msg.File)
 	}
 
-	lineExpected := 261 // Update this if code is updated above
+	lineExpected := 258 // Update this if code is updated above
 	if msg.Line != lineExpected {
 		t.Errorf("msg.Line: expected %d, got %d", lineExpected, msg.Line)
 	}
@@ -389,7 +386,7 @@ func TestReportCallerEnabled(t *testing.T) {
 		t.Error("_line dowes not have the correct type")
 	}
 
-	lineExpected := 359 // Update this if code is updated above
+	lineExpected := 356 // Update this if code is updated above
 	if msg.Line != lineExpected {
 		t.Errorf("msg.Extra[\"_line\"]: expected %d, got %d", lineExpected, int(lineGot))
 	}

